tt_metal/impl/device/device.hpp

```diff
@@ -53,6 +53,14 @@ static constexpr float  EPS_GS = 0.001953125f;
 static constexpr float  EPS_WHB0 = 1.19209e-7f;
 static constexpr float  EPS_BH = EPS_WHB0;
 
+static constexpr float  NAN_GS = 6.9752e19;
+static constexpr float  NAN_WHB0 = 7.0040e+19;
+static constexpr float  NAN_BH = NAN_WHB0;
+
+static constexpr float  INF_GS = 1.6948e38;
+static constexpr float  INF_WHB0 = 1.7014e+38;
```

Comment:
How did you find this number?
I mean INF_WHB0.

I set every element of both bfloat16 tiles A and B to the max value and add two tiles.
Then, what I get is 0x7FB0 in every element of the output tile which equals to the PyTorch output and [this](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format#Positive_and_negative_infinity).

Reply:
If I print INF_WHB0 in hexadecimal, I get 7EFFFF8B.

Reply:
@umadevimcw Any comment about this?

Reply:
https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/Handling_Special_Value/special_values.md#representation
I get the same values noted in this page.
I don't know how the above numbers in the file (_tt_metal/impl/device/device.hpp_) can be generated.

@ttmtrajkovic Can you give any advice?

Reply:
@umadevimcw
Would you tell me how can I reproduce the output you got?

Reply:
@razorback3 I ran the test mentioned in this issue #6720 #6722 (as mentioned in the description on this PR) and observed the values updated in the device.hpp file. I didn't do any changes i ran the tests mentioned in the issues

Reply:
@razorback3,
The NaN values from device.hpp are incorrect, I am not sure where did they come from but it’s possible that they got added before I got to define them through the tech report.
I will find the owner of this file and update this, but in the meantime, please update locally to match values I’ve specified.

Reply:
@razorback3 
https://github.com/tenstorrent/tt-metal/issues/8945#issuecomment-2146247945

As discussed here developer/user have to handle the storage of Nan /inf due to hardware limitations.
Hence we have used nan_to_num to compare Nan to numbers. As mentioned in earlier comments special value doc are updated recently ( got into repo after this PR) updating param with correct values is appropriate one. Will update it in separate PR